src/dft/descriptor.cxx

```diff
@@ -26,39 +27,74 @@ namespace detail {
 
 template <precision prec, domain dom>
 void descriptor<prec, dom>::set_value(config_param param, ...) {
-    int err = 0;
     va_list vl;
     va_start(vl, param);
     switch (param) {
-        case config_param::INPUT_STRIDES: [[fallthrough]];
+        case config_param::FORWARD_DOMAIN:
+            throw mkl::invalid_argument("DFT", "set_value", "Read-only parameter.");
+            break;
+        case config_param::DIMENSION:
+            throw mkl::invalid_argument("DFT", "set_value", "Read-only parameter.");
+            break;
+        case config_param::LENGTHS: {
+            if (values_.rank == 1) {
```

Comment:
Would it make sense to have a check here for invalid dimensions like 0? While writing the test, I did this mistake and the error is a very ugly crash on commit method

Reply:
Dimensions should always be > 0, and is read-only. I've added a check in the constructor in [0b9d613](https://github.com/hjabird/oneMKL/pull/3/commits/0b9d61393167e2fcb17367e7a0decdf280c85f73), so this should always be the case here.

Reply:
Sorry, I mean dimensions as in Lengths not rank. It's possible to do set_value(LENGTHS, {0,0,0}) or something like that. A check on the constructor for that would also make sense I think

Reply:
Checks added in [0739bf1](https://github.com/hjabird/oneMKL/pull/3/commits/0739bf1142250e869b2ae1991dd1f44066203ed0).